pkg/blockchain/interface.go

```diff
@@ -27,3 +28,12 @@ type ChainClient interface {
 	ethereum.LogFilterer
 	ethereum.ChainIDReader
 }
+
+type IGroupMessagePublisher interface {
```

Comment:
Just to understand better, what are some other implementations of this interface that might be possible in the future?

Reply:
The main reason I like creating interfaces for everything is that it makes everything mockable for tests.